File: ruby/singular_to_plural.rb
puts 'Please enter the singular word:'
word = gets.chomp

word.match('[s|sh|ch|o|x]$') do |m|
  puts "#{word}es"
  exit
end

word.match('(.*)(fe?)') do |m|
  puts "#{m[1]}ves"
  exit
end

word.match('(.*[^aiueo])y$') do |m|
  puts "#{m[1]}ies"
  exit
end

puts "#{word}s"

File: elixir/singular_to_plural.ex
word = String.trim(IO.gets "Please enter the singular word:")

if String.match?(word, ~r/[s|sh|ch|o|x]$/) do
  IO.puts word <> "es"
  System.halt(0)
end

matched = Regex.run(~r/(.*)(fe?)$/, word)
if matched do
  IO.puts Enum.at(matched, 1) <> "ves"
  System.halt(0)
end

matched = Regex.run(~r/(.*[^aiueo])y$/, word)
if matched do
  IO.puts Enum.at(matched, 1) <> "ies"
  System.halt(0)
end

IO.puts word <> "s"
